src/python/pants/option/arg_splitter_test.py

```diff
@@ -293,15 +287,25 @@ def help_no_arguments_test(command_line: str, **expected):
     "command_line, expected",
     [
         help_no_arguments_test("./pants"),
-        help_no_arguments_test("./pants help"),
-        help_no_arguments_test("./pants -h"),
-        help_no_arguments_test("./pants --help"),
-        help_no_arguments_test("./pants help-advanced", expected_help_advanced=True),
-        help_no_arguments_test("./pants help --help-advanced", expected_help_advanced=True),
-        help_no_arguments_test("./pants --help-advanced", expected_help_advanced=True),
-        help_no_arguments_test("./pants --help --help-advanced", expected_help_advanced=True),
-        help_no_arguments_test("./pants --help-advanced --help", expected_help_advanced=True),
-        help_no_arguments_test("./pants help-all", expected_help_all=True),
+        help_no_arguments_test("./pants help", "help"),
+        help_no_arguments_test("./pants -h", "help"),
+        help_no_arguments_test("./pants --help", "help"),
+        help_no_arguments_test(
+            "./pants help-advanced", "help-advanced", expected_help_advanced=True
+        ),
+        help_no_arguments_test(
+            "./pants help --help-advanced", "help", "help-advanced", expected_help_advanced=True
```

Comment:
Knowing now how flag scoping works, I would've expected this to work too, alas it does not (not now nor before):
`./pants help --advanced ...`

I think this could be worth looking at fixing...